### PR TITLE
Security Question verify and testcafe test

### PR DESCRIPTION
**Pull request by haishengwu-okta**
## Description:

- update authenticator object structure
- also pull latest idxjs
- add verify SQ page
- add test for SQ enroll and verify

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:

hw-authenticator-api-sq

### Issue:

- [OKTA-304275](https://oktainc.atlassian.net/browse/OKTA-304275)

### CODE REVIEW

File: playground/mocks/data/idp/idx/authenticator-select-verify-options.json [hunk 1]
```diff
@@ -84,6 +84,38 @@
                     ]
                   }
                 }
+              },
+              {
+                "label": "Okta Phone",
```
Comment (magizh-okta) [inline playground/mocks/data/idp/idx/authenticator-select-verify-options.json:89]: This doesnt have display name like others?

File: test/testcafe/spec/EnrollAuthenticatorSecurityQuestion_spec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  await challengeFactorPage.navigateToPage();
+  return challengeFactorPage;
+}
+
+test.requestHooks(answerRequestLogger, authenticatorRequiredSecurityQuestionMock)(`enroll security question`, async t => {
+  const challengeFactorPageObject = await setup(t);
+
+  const radioOptionLabel = await challengeFactorPageObject.clickChooseSecurityQuestion();
+  await t.expect(radioOptionLabel).eql('Choose a security question');
+  await challengeFactorPageObject.selectSecurityQuestion(1);
```
Comment (magizh-okta) [inline test/testcafe/spec/EnrollAuthenticatorSecurityQuestion_spec.js:36]: Can we add a check to make sure the non relevant fields are hidden in both the tests. 

eg) In this test the custom question field should be hidden,

Comment (haishengwu-okta) [reply test/testcafe/spec/EnrollAuthenticatorSecurityQuestion_spec.js:36]: Good point! Thanks!

File: src/v2/view-builder/views/security-question/ChallengeAuthenticatorSecurityQuestion.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  save () {
+    return loc('oie.verify.button', 'login');
+  },
+
+  getUISchema () {
+    const uiSchemas = BaseForm.prototype.getUISchema.apply(this, arguments);
+    const questionKey = uiSchemas.filter(s => s.name.indexOf('questionKey') >= 0);
+    const answer = uiSchemas.filter(s => s.name.indexOf('answer') >= 0);
+
+    if (questionKey.length === 1 && answer.length === 1) {
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/views/security-question/ChallengeAuthenticatorSecurityQuestion.js:19]: Can we add a comment explaining this check ? I am guessing this is to show the question label above the answer input field ?

Comment (haishengwu-okta) [reply src/v2/view-builder/views/security-question/ChallengeAuthenticatorSecurityQuestion.js:19]: yes. will add comment.

**Pull request merged**